src/big/big_decimal.cr

```diff
@@ -235,7 +235,7 @@ struct BigDecimal < Number
   # BigDecimal.new(1).div(BigDecimal.new(2))    # => BigDecimal(@value=5, @scale=2)
   # BigDecimal.new(1).div(BigDecimal.new(3), 5) # => BigDecimal(@value=33333, @scale=5)
   # ```
-  def div(other : BigDecimal, precision = DEFAULT_PRECISION) : BigDecimal
+  def div(other : BigDecimal, precision : UInt64 | Int32 = DEFAULT_PRECISION) : BigDecimal
```

Comment:
I think any unions exclusively consisting of integral types should just be `Int::Primitive`

There's no real reason for these two types to be mentioned explicitly

Reply:
Oh, good call, will make that change.

Reply:
I don't think `Int::Primitive` is a good type for that. Most of the time it should be either `Int` or the biggest supported integer type (making use of autocasting).

Reply:
I can't think of a reason where a negative value would be provided, so `UInt64`?

Reply:
Negative values might not make sense, but even a positive value can be of a signed integer type. And that wouldn't autocast to an unsigned one.
`Int` should be fine here.